Build error on OS X

I downloaded the master branch minutes ago and built on OS X El Capitan. Compilation failed on this error:

```
In file included from ../src/lib_ccx/es_userdata.c:2:
../src/lib_ccx/ccx_decoders_vbi.h:4:10: fatal error: 'zvbi_decoder.h' file not found
#include <zvbi_decoder.h>
```

Here is the full output:

```
$ ./build.command
../src/lib_ccx/ccx_decoders_isdb.c:373:12: warning: assigning to 'char *' from 'unsigned char *' converts between pointers to integer types with different
      sign [-Wpointer-sign]
        text->buf = ptr;
                  ^ ~~~
../src/lib_ccx/ccx_decoders_isdb.c:376:1: warning: control may reach end of non-void function [-Wreturn-type]
}
^
../src/lib_ccx/ccx_decoders_isdb.c:515:32: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                                if (ccx_strstr_ignorespace(text->buf, sb_text->buf))
                                                           ^~~~~~~~~
../src/lib_ccx/ccx_decoders_isdb.c:462:56: note: passing argument to parameter 'str1' here
static int ccx_strstr_ignorespace(const unsigned char *str1, const unsigned char *str2)
                                                       ^
../src/lib_ccx/ccx_decoders_isdb.c:515:43: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                                if (ccx_strstr_ignorespace(text->buf, sb_text->buf))
                                                                      ^~~~~~~~~~~~
../src/lib_ccx/ccx_decoders_isdb.c:462:83: note: passing argument to parameter 'str2' here
static int ccx_strstr_ignorespace(const unsigned char *str1, const unsigned char *str2)
                                                                                  ^
../src/lib_ccx/ccx_decoders_isdb.c:519:34: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                                        if (!ccx_strstr_ignorespace(sb_text->buf, text->buf))
                                                                    ^~~~~~~~~~~~
../src/lib_ccx/ccx_decoders_isdb.c:462:56: note: passing argument to parameter 'str1' here
static int ccx_strstr_ignorespace(const unsigned char *str1, const unsigned char *str2)
                                                       ^
../src/lib_ccx/ccx_decoders_isdb.c:519:48: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                                        if (!ccx_strstr_ignorespace(sb_text->buf, text->buf))
                                                                                  ^~~~~~~~~
../src/lib_ccx/ccx_decoders_isdb.c:462:83: note: passing argument to parameter 'str2' here
static int ccx_strstr_ignorespace(const unsigned char *str1, const unsigned char *str2)
                                                                                  ^
../src/lib_ccx/ccx_decoders_isdb.c:544:32: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                                if (ccx_strstr_ignorespace(text->buf, sb_text->buf))
                                                           ^~~~~~~~~
../src/lib_ccx/ccx_decoders_isdb.c:462:56: note: passing argument to parameter 'str1' here
static int ccx_strstr_ignorespace(const unsigned char *str1, const unsigned char *str2)
                                                       ^
../src/lib_ccx/ccx_decoders_isdb.c:544:43: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                                if (ccx_strstr_ignorespace(text->buf, sb_text->buf))
                                                                      ^~~~~~~~~~~~
../src/lib_ccx/ccx_decoders_isdb.c:462:83: note: passing argument to parameter 'str2' here
static int ccx_strstr_ignorespace(const unsigned char *str1, const unsigned char *str2)
                                                                                  ^
../src/lib_ccx/ccx_decoders_isdb.c:1312:24: warning: passing 'unsigned char [1024]' to parameter of type 'char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
                add_cc_sub_text(sub, buffer, ctx->prev_timestamp, ctx->timestamp, "NA", "ISDB", CCX_ENC_UTF_8);
                                     ^~~~~~
../src/lib_ccx/ccx_common_common.h:47:52: note: passing argument to parameter 'str' here
int add_cc_sub_text(struct cc_subtitle *sub, char *str, LLONG start_time,
                                                   ^
9 warnings generated.
In file included from ../src/lib_ccx/ccx_decoders_vbi.c:1:
../src/lib_ccx/ccx_decoders_vbi.h:4:10: fatal error: 'zvbi_decoder.h' file not found
#include <zvbi_decoder.h>
         ^
1 error generated.
../src/lib_ccx/ccx_encoders_common.c:139:12: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                        if (c < 256)
                                            ~ ^ ~~~
../src/lib_ccx/ccx_encoders_common.c:154:12: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                        if (c < 256)
                                            ~ ^ ~~~
../src/lib_ccx/ccx_encoders_common.c:170:12: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                        if (c < 256)
                                            ~ ^ ~~~
../src/lib_ccx/ccx_encoders_common.c:190:12: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                        if (c < 256)
                                            ~ ^ ~~~
4 warnings generated.
../src/lib_ccx/ccx_encoders_helpers.c:111:11: warning: enumeration value 'CCX_ENC_ASCII' not handled in switch [-Wswitch]
                switch (ctx->encoding)
                        ^
../src/lib_ccx/ccx_encoders_helpers.c:248:11: warning: enumeration value 'CCX_ENC_ASCII' not handled in switch [-Wswitch]
                switch (ctx->encoding)
                        ^
2 warnings generated.
../src/lib_ccx/ccx_gxf.c:419:68: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                result = buffered_read(demux, (unsigned char*)ctx->media_name, MIN(tag_len, STR_LEN));
                                                                                               ^   ~~~~~~~  ~~~~~~~
../src/lib_ccx/ccx_gxf.c:23:24: note: expanded from macro 'MIN'
#define MIN(a, b) ( (a < b) ? a : b)
                     ~ ^ ~
../src/lib_ccx/ccx_gxf.c:421:19: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                if (result != MIN(tag_len, STR_LEN) )
                                              ^   ~~~~~~~  ~~~~~~~
../src/lib_ccx/ccx_gxf.c:23:24: note: expanded from macro 'MIN'
#define MIN(a, b) ( (a < b) ? a : b)
                     ~ ^ ~
../src/lib_ccx/ccx_gxf.c:426:17: warning: comparison of constant 256 with expression of type 'unsigned char' is always false
      [-Wtautological-constant-out-of-range-compare]
                                if (tag_len > STR_LEN)
                                    ~~~~~~~ ^ ~~~~~~~
../src/lib_ccx/ccx_gxf.c:536:74: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                result = buffered_read(demux, (unsigned char*)vid_track->track_name, MIN(tag_len, STR_LEN));
                                                                                                     ^   ~~~~~~~  ~~~~~~~
../src/lib_ccx/ccx_gxf.c:23:24: note: expanded from macro 'MIN'
#define MIN(a, b) ( (a < b) ? a : b)
                     ~ ^ ~
../src/lib_ccx/ccx_gxf.c:538:19: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                if (result != MIN(tag_len, STR_LEN) )
                                              ^   ~~~~~~~  ~~~~~~~
../src/lib_ccx/ccx_gxf.c:23:24: note: expanded from macro 'MIN'
#define MIN(a, b) ( (a < b) ? a : b)
                     ~ ^ ~
../src/lib_ccx/ccx_gxf.c:543:17: warning: comparison of constant 256 with expression of type 'unsigned char' is always false
      [-Wtautological-constant-out-of-range-compare]
                                if (tag_len > STR_LEN)
                                    ~~~~~~~ ^ ~~~~~~~
../src/lib_ccx/ccx_gxf.c:611:73: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                result = buffered_read(demux, (unsigned char*)ad_track->track_name, MIN(tag_len, STR_LEN));
                                                                                                    ^   ~~~~~~~  ~~~~~~~
../src/lib_ccx/ccx_gxf.c:23:24: note: expanded from macro 'MIN'
#define MIN(a, b) ( (a < b) ? a : b)
                     ~ ^ ~
../src/lib_ccx/ccx_gxf.c:613:19: warning: comparison of constant 256 with expression of type 'unsigned char' is always true
      [-Wtautological-constant-out-of-range-compare]
                                if (result != MIN(tag_len, STR_LEN) )
                                              ^   ~~~~~~~  ~~~~~~~
../src/lib_ccx/ccx_gxf.c:23:24: note: expanded from macro 'MIN'
#define MIN(a, b) ( (a < b) ? a : b)
                     ~ ^ ~
../src/lib_ccx/ccx_gxf.c:618:17: warning: comparison of constant 256 with expression of type 'unsigned char' is always false
      [-Wtautological-constant-out-of-range-compare]
                                if (tag_len > STR_LEN)
                                    ~~~~~~~ ^ ~~~~~~~
9 warnings generated.
In file included from ../src/lib_ccx/es_userdata.c:2:
../src/lib_ccx/ccx_decoders_vbi.h:4:10: fatal error: 'zvbi_decoder.h' file not found
#include <zvbi_decoder.h>
         ^
1 error generated.
../src/lib_ccx/file_functions.c:219:27: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
                if (ctx->filebuffer_pos < 0)
                    ~~~~~~~~~~~~~~~~~~~ ^ ~
../src/lib_ccx/file_functions.c:222:52: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
                        if ((ctx->filebuffer_pos + ctx->startbytes_pos) < 0)
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~
../src/lib_ccx/file_functions.c:346:23: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
                                                        if (op + bytes < 0) // Would mean moving beyond start of file: Not supported
                                                            ~~~~~~~~~~ ^ ~
../src/lib_ccx/file_functions.c:441:19: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
                        if (op + bytes < 0) // Would mean moving beyond start of file: Not supported
                            ~~~~~~~~~~ ^ ~
4 warnings generated.
../src/lib_ccx/networking.c:124:38: warning: passing 'const unsigned char *' to parameter of type 'const char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
        if (write_block(srv_sd, BIN_HEADER, data, len) <= 0)
                                            ^~~~
../src/lib_ccx/networking.c:61:55: note: passing argument to parameter 'buf' here
ssize_t write_block(int fd, char command, const char *buf, size_t buf_len);
                                                      ^
../src/lib_ccx/networking.c:148:36: warning: passing 'const unsigned char *' to parameter of type 'const char *' converts between pointers to integer types
      with different sign [-Wpointer-sign]
        if (write_block(srv_sd, BIN_DATA, data, len) <= 0)
                                          ^~~~
../src/lib_ccx/networking.c:61:55: note: passing argument to parameter 'buf' here
ssize_t write_block(int fd, char command, const char *buf, size_t buf_len);
                                                      ^
2 warnings generated.
../src/lib_ccx/output.c:21:1: warning: control reaches end of non-void function [-Wreturn-type]
}
^
1 warning generated.
../src/lib_ccx/params_dump.c:108:10: warning: enumeration value 'CCX_ENC_ASCII' not handled in switch [-Wswitch]
        switch (ccx_options.enc_cfg.encoding)
                ^
1 warning generated.
../src/lib_ccx/ts_functions.c:61:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_PES;
                ~~~~~~ ^~~~~~~
../src/lib_ccx/ts_functions.c:65:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_H264;
                ~~~~~~ ^~~~~~~~
../src/lib_ccx/ts_functions.c:69:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_DVB_SUBTITLE;
                ~~~~~~ ^~~~~~~~~~~~~~~~
../src/lib_ccx/ts_functions.c:73:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_ISDB_SUBTITLE;
                ~~~~~~ ^~~~~~~~~~~~~~~~~
../src/lib_ccx/ts_functions.c:77:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_HAUPPAGE;
                ~~~~~~ ^~~~~~~~~~~~
../src/lib_ccx/ts_functions.c:81:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_TELETEXT;
                ~~~~~~ ^~~~~~~~~~~~
../src/lib_ccx/ts_functions.c:85:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_PRIVATE_MPEG2_CC;
                ~~~~~~ ^~~~~~~~~~~~~~~~~~~~
../src/lib_ccx/ts_functions.c:89:10: warning: implicit conversion from enumeration type 'enum ccx_bufferdata_type' to different enumeration type
      'enum ccx_stream_type' [-Wenum-conversion]
                return CCX_PES;
                ~~~~~~ ^~~~~~~
../src/lib_ccx/ts_functions.c:406:24: warning: implicit conversion from enumeration type 'enum ccx_stream_type' to different enumeration type
      'enum ccx_bufferdata_type' [-Wenum-conversion]
        ptr->bufferdatatype = get_buffer_type(cinfo);
                            ~ ^~~~~~~~~~~~~~~~~~~~~~
9 warnings generated.
../src/lib_ccx/ts_tables_epg.c:463:8: warning: initializing 'char *' with an expression of type 'uint8_t *' (aka 'unsigned char *') converts between
      pointers to integer types with different sign [-Wpointer-sign]
        char *dp = &decode_buffer[0];
              ^    ~~~~~~~~~~~~~~~~~
../src/lib_ccx/ts_tables_epg.c:473:10: warning: returning 'uint8_t *' (aka 'unsigned char *') from a function with result type 'char *' converts between
      pointers to integer types with different sign [-Wpointer-sign]
                return decode_buffer;
                       ^~~~~~~~~~~~~
../src/lib_ccx/ts_tables_epg.c:604:15: warning: passing 'uint8_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to
      integer types with different sign [-Wpointer-sign]
        osize=strlen(decode_buffer);
                     ^~~~~~~~~~~~~
/usr/include/string.h:82:28: note: passing argument to parameter here
size_t   strlen(const char *);
                            ^
../src/lib_ccx/ts_tables_epg.c:611:9: warning: returning 'uint8_t *' (aka 'unsigned char *') from a function with result type 'char *' converts between
      pointers to integer types with different sign [-Wpointer-sign]
        return out;
               ^~~
../src/lib_ccx/ts_tables_epg.c:682:23: warning: assigning to 'char *' from 'uint8_t *' (aka 'unsigned char *') converts between pointers to integer types
      with different sign [-Wpointer-sign]
                event->extended_text=net;
                                    ^~~~
../src/lib_ccx/ts_tables_epg.c:692:12: warning: initializing 'uint8_t *' (aka 'unsigned char *') with an expression of type 'char *' converts between
      pointers to integer types with different sign [-Wpointer-sign]
                uint8_t *old = event->extended_text;
                         ^     ~~~~~~~~~~~~~~~~~~~~
../src/lib_ccx/ts_tables_epg.c:693:48: warning: passing 'char *' to parameter of type 'uint8_t *' (aka 'unsigned char *') converts between pointers to
      integer types with different sign [-Wpointer-sign]
                event->extended_text = EPG_DVB_decode_string(event->extended_text, strlen(event->extended_text));
                                                             ^~~~~~~~~~~~~~~~~~~~
../src/lib_ccx/ts_tables_epg.c:458:38: note: passing argument to parameter 'in' here
char* EPG_DVB_decode_string(uint8_t *in, size_t size)
                                     ^
7 warnings generated.
../src/lib_ccx/utility.c:484:24: warning: comparison of constant 65536 with expression of type 'unsigned short' is always true
      [-Wtautological-constant-out-of-range-compare]
        } else if (utf16_char < 0x010000) {
                   ~~~~~~~~~~ ^ ~~~~~~~~
../src/lib_ccx/utility.c:489:24: warning: comparison of constant 1114112 with expression of type 'unsigned short' is always true
      [-Wtautological-constant-out-of-range-compare]
        } else if (utf16_char < 0x110000) {
                   ~~~~~~~~~~ ^ ~~~~~~~~
2 warnings generated.
../src/zlib/inflate.c:1470:61: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    if (strm == Z_NULL || strm->state == Z_NULL) return -1L << 16;
                                                        ~~~ ^
1 warning generated.

```

--- a/mac/build.command
+++ b/mac/build.command
@@ -1,12 +1,15 @@
 #!/bin/bash
 cd `dirname $0`
 BLD_FLAGS="-std=gnu99 -Wno-write-strings -DGPAC_CONFIG_DARWIN -D_FILE_OFFSET_BITS=64 -Dfopen64=fopen -Dopen64=open -Dlseek64=lseek"
-BLD_INCLUDE="-I../src/lib_ccx/ -I../src/gpacmp4/ -I../src/libpng/ -I../src/zlib/"
-SRC_LIBPNG="$(find ../src/libpng/ -name '*.c')"
-SRC_ZLIB="$(find ../src/zlib/ -name '*.c')"
-SRC_CCX="$(find ../src/lib_ccx/ -name '*.c')"
-SRC_GPAC="$(find ../src/gpacmp4/ -name '*.c')"
-BLD_SOURCES="../src/ccextractor.c $SRC_CCX $SRC_GPAC $SRC_ZLIB $SRC_LIBPNG"
+BLD_INCLUDE="-I../src/lib_ccx -I../src/gpacmp4 -I../src/libpng -I../src/zlib -I../src/zvbi"
+SRC_LIBPNG="$(find ../src/libpng -name '*.c')"
+SRC_ZLIB="$(find ../src/zlib -name '*.c')"
+SRC_ZVBI="$(find ../src/zvbi -name '*.c')"
+SRC_CCX="$(find ../src/lib_ccx -name '*.c')"
+SRC_GPAC="$(find ../src/gpacmp4 -name '*.c')"
+BLD_SOURCES="../src/ccextractor.c $SRC_CCX $SRC_GPAC $SRC_ZVBI $SRC_ZLIB $SRC_LIBPNG"
 BLD_LINKER="-lm -liconv"
 
 gcc $BLD_FLAGS $BLD_INCLUDE -o ccextractor $BLD_SOURCES $BLD_LINKER
+
+

